### PR TITLE
Fixes for Test-DbaDbDataMaskingConfig

### PR DESCRIPTION
**Pull request by BartekR**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes # )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [x] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Data masking config file created with `New-DbaDbMaskingConfig` returns errors when tested with `Test-DbaDbDataMaskingConfig`. This fixes the following problems:

1. `tinyint` and `float` columns are valid. Before the fix I had _ColumnType is not a supported data type_ errors

```
Test-DbaDbDataMaskingConfig -FilePath D:\temp\_mssql\DataMaskingConfig.json

Table Column Value   Error
----- ------ -----   -----
Test2 id1    tinyint ColumnType is not a supported data type
Test2 id2    float   ColumnType is not a supported data type
```

2. Added _Finance_ masking type for Creditcard Discover (it was empty)
3. Changed _Category_ subtype to _Categories_ - to make it coherent with `bin\randomizer\en.randomizertypes.csv`. Solves problems like _Category SubType is not valid_:

```
Test-DbaDbDataMaskingConfig -FilePath D:\temp\_mssql\WideWorldImporters.DataMaskingConfig.json                 

Table                  Column          Value    Error
-----                  ------          -----    -----
People                 IsSalesperson   Category SubType is not valid
People_Archive         IsSalesperson   Category SubType is not valid
StateProvinces         SalesTerritory  Category SubType is not valid
StateProvinces_Archive SalesTerritory  Category SubType is not valid
```

**WARNING: This might introduce a breaking change**, that's why I checked _Breaking change_ option. 

### Approach

- Added `tinyint` and `float` to `$supportedDataTypes`
- Added _Finance_ MaskingType for Creditcard Discover (`pii-patterns.json`)
- Changed _Category_ subtype to _Categories_ (`pii-knownnames.json`)
- Added tests to check if masking types and subtypes exist in `en.randomizertypes.csv`

Tested the new Pester tests (all tests pass in PowerShell 5.1, Windows 10; _Context Validate parameters_ does not pass in Pester 5.0.4, PowerShell 7.0.3, Windows 10)

```
.\tests\manual.pester.ps1 -Path .\tests\Test-DbaDbDataMaskingConfig.Tests.ps1
Pester v4.10.1
Executing all tests in 'C:\Users\BartekR\source\bartekr_github\dbatools\tests\Test-DbaDbDataMaskingConfig.Tests.ps1'

Executing script C:\Users\BartekR\source\bartekr_github\dbatools\tests\Test-DbaDbDataMaskingConfig.Tests.ps1
Running C:\Users\BartekR\source\bartekr_github\dbatools\tests\Test-DbaDbDataMaskingConfig.Tests.ps1

  Describing Test-DbaDbDataMaskingConfig Unit Tests

    Context Validate parameters
      [+] Should only contain our specific parameters 15ms

    Context PII Known Names
      [+] All masking types match randomizer types 6ms
      [+] All masking subtypes match randomizer subtypes 5ms

    Context PII patterns
      [+] All masking types match randomizer types 6ms
      [+] All masking subtypes match randomizer subtypes 7ms
Tests completed in 269ms
Tests Passed: 5, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0
```

### CODE REVIEW

**Comment by potatoqualitee** [comment]
Thanks Bartek! I shall wait for @sanderstad's approval 💯

**Review by sanderstad — APPROVED** [review]
Approved! Thanks for looking into this

**Comment by potatoqualitee** [comment]
🍌 👯 ! thank you both

**Pull request merged**
